### PR TITLE
Tighten up proxy to only allow opensafely images.

### PR DESCRIPTION
**Pull request by bloodearnest**
Returns a 404 for other organisation images in ghcr.io.

### CODE REVIEW

File: default.conf.template [hunk 1]
```diff
@@ -5,45 +5,67 @@
...
     server_name ${PROXY_DOMAIN} ${PROXY_ORIGIN};
     root /var/www/html;
     listen ${PORT};
 
     # we need a resolver configured so we can look up domains when redirecting
     resolver ${RESOLVER};
 
-    location / {
-        # Only GET/HEAD
+    # excat match, initial query, rewrite www-authentication header
```
Comment (madwort) [inline default.conf.template:30]: typo `exact`

File: default.conf.template [hunk 2]
```diff
@@ -5,45 +5,67 @@
...
+        # passing back to the client
+        proxy_intercept_errors on;
+        error_page 301 302 307 = @handle_redirect;
+    }
+
     location @handle_redirect {
         # set saves the initial response's Location: header
         set $redirect '$upstream_http_location';
-        # strip the ghcr.io auth or else AWS rejects it
+        # strip the ghcr.io auth or else AWS rejects it, also stops is leaking
```
Comment (madwort) [inline default.conf.template:60]: typo `is` -> `it`

File: default.conf.template [hunk 3]
```diff
@@ -5,45 +5,67 @@
...
     root /var/www/html;
     listen ${PORT};
 
     # we need a resolver configured so we can look up domains when redirecting
     resolver ${RESOLVER};
 
-    location / {
-        # Only GET/HEAD
+    # excat match, initial query, rewrite www-authentication header
+    location = /v2/ {
```
Comment (madwort) [inline default.conf.template:31]: does this mean that `limit_except` only applies to exactly `/v2/`, and not to anything else (e.g. `/v2/opensafely/blah/`)?

**Review by madwort — CHANGES_REQUESTED** [review]

File: default.conf.template (hunk 3 above)
Comment (bloodearnest) [reply default.conf.template:31]: the limit_except clause is repeated for each explicit location, so it does apply for /v2/opensafely/blah, however I think you have a point that it doesn't apply to /v2/other-org, which I'll add.

**Review by madwort — APPROVED** [review]

**Pull request merged**